### PR TITLE
MSR - Update A8 to support queen wall removal

### PR DESCRIPTION
**Pull request by dyceron**
Also fix Optional Patches not persisting

### CODE REVIEW

File: randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py [hunk 1]
```diff
@@ -29,7 +31,9 @@ def __init__(self, editor: PresetEditor, game_description: GameDescription, wind
         super().__init__(editor, game_description, window_manager)
         self.setupUi(self)
 
-        self.setCentralWidget(self.root_widget)
```
Comment (ThanatosGit) [inline randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py:32]: Why was the line removed?

File: randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py [hunk 2]
```diff
@@ -39,6 +43,13 @@ def tab_title(cls) -> str:
     def uses_patches_tab(cls) -> bool:
         return True
 
+    def _add_persist_option(self, check: QtWidgets.QCheckBox, attribute_name: str):
```
Comment (ThanatosGit) [inline randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py:46]: add the return type

File: randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py [hunk 3]
```diff
@@ -39,6 +43,13 @@ def tab_title(cls) -> str:
...
+        def persist(value: bool):
```
Comment (ThanatosGit) [inline randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py:47]: add the return type

File: randovania/gui/ui_files/preset_msr_patches.ui [hunk 4]
```diff
@@ -180,6 +180,26 @@
...
+             </property>
+            </widget>
+           </item>
+           <item>
+            <widget class="QLabel" name="reverse_area8_label">
+             <property name="mouseTracking">
+              <bool>true</bool>
+             </property>
+             <property name="text">
+              <string>Removes the wall between the Queen and Baby, allowing for earlier access into Area 8 by enerting the area backwards</string>
```
Comment (ThanatosGit) [inline randovania/gui/ui_files/preset_msr_patches.ui:196]: `enerting` is not in my vocabulary and also don't show up in my dictionary. `entering` ?

**Review by ThanatosGit — CHANGES_REQUESTED** [review]

File: randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py (hunk 1 above)
Comment (dyceron) [reply randovania/games/samus_returns/gui/preset_settings/msr_patches_tab.py:32]: It didn't seem to make a difference either way.

**Review by ThanatosGit — APPROVED** [review]

**Pull request merged**